lib/mongoose_push/application.ex

```diff
@@ -6,22 +6,37 @@ defmodule MongoosePush.Application do
   use Application
   require Logger
 
+  @typedoc "Possible keys in both services"
+  @type fcm_key :: :key | :pool_size | :mode | :endpoint
+  @type apns_key :: :cert | :key | :pool_size | :mode | :endpoint | :use_2197
+
+  @typedoc """
+  In FCM `:key` and `:pool_size` are required and `:mode` has to be either `:dev` or `:prod`
+  """
+  @type fcm_config :: [fcm_key: String.t() | atom | integer]
+  @typedoc """
+  In APNS `:cert`, `:key` and `:pool_size` are required. `:mode` has to be either `:dev` or `:prod`
+  """
+  @type apns_config :: [apns_key: String.t() | atom | integer]
+
+  @type pool_info :: {atom(), fcm_config | apns_config}
```

Comment:
What is the `atom()` for? If this is a pool name, we should have a type 
`pool_name() :: atom()`. Typespecs are part of documentation, and the names matter, sometimes even more then actual definition.